client/src/pages/RoomPage/index.tsx

```diff
@@ -22,34 +29,39 @@ const notificationColors = {
   leave: { color: "danger", icon: "Info" },
 };
 
-
 function RoomPage({ name }: { name: string }): JSX.Element {
   const [announcement, setAnnouncement] = useState("");
   const [showNotification, setShowNotification] = useState(false);
   const [notificationTheme, setNotificationTheme] = useState("join");
-  const [isAutoSleepDisabled, setAutoSleepDisabled] = useState(false)
+  const [isAutoSleepDisabled, setAutoSleepDisabled] = useState(false);
   const { socket } = useContext(SocketContext);
   const { stream, setStream } = useContext(DeviceContext);
   const [visualizer, setVisualizer] = useState(null as unknown as AudioVisualizer);
   const visualizerRef = useRef(visualizer);
-  const [selfUserInfo, setSelfUserInfo] = useState<UserInfo>({ ...defaultUserInfo, name });
-  const selfUserInfoRef = useRef(selfUserInfo);
   const { userInfos, addUserInfo, removeUserInfo } = useContext(UserInfoContext);
-  const userInfosRef = useRef(userInfos)
+  const selfInfoRef = useRef(null as unknown as UserInfo);
+  const userInfosRef = useRef(userInfos);
   const history = useHistory();
   const [showModal, setShowModal] = useState(false);
   const networkRef = useRef(null as unknown as Network);
-  const [ soundEffectPlayer, setSoundEffectPlayer ] = useState<HTMLAudioElement | null>(null);
+  const [soundEffectPlayer, setSoundEffectPlayer] = useState<HTMLAudioElement | null>(null);
+
+  const selfInfo = useMemo(() => userInfosRef.current[socket.id], [userInfosRef.current[socket.id]]);
 
   // when new input is selected update all tracks and send a new offer out
   const onSelect = (_stream) => {
     setStream(_stream);
   };
 
+  // Using ref here for the non-rerendered AudioVisualizer class
+  // To-do: rf-369
   const updateSelfUserInfo = (info) => {
```

Comment:
```
  const updateSelfInfo = (info: Partial<UserInfo>) => {
```

If all the occurrences of selfUserInfo has been changed to selfInfo, maybe change function names accordingly as well.

Also, would be nice to have type annotation

Reply:
Seems to affect other files as well, might be easier to keep  the variable name `selfInfo` as `selfUserInfo`

Reply:
I agree I'll make these changes.